### PR TITLE
Added close method to IdentityPopup r=whimboo

### PR DESCRIPTION
**Pull request by Ethcelon**
This is required for tests using identity popup
#99

### CODE REVIEW

File: firefox_puppeteer/ui/toolbars.py [hunk 1]
```diff
@@ -343,6 +343,25 @@ def selected_index(self):
 class IdentityPopup(BaseLib):
     """Wraps DOM elements and methods for interacting with the identity popup."""
 
+    def close(self, force=False):
```
Comment (whimboo) [inline firefox_puppeteer/ui/toolbars.py]: Please add methods alphabetically after the properties.

File: firefox_puppeteer/ui/toolbars.py [hunk 2]
```diff
@@ -343,6 +343,25 @@ def selected_index(self):
...
+        if not self.is_open:
+            return
+
+        if force:
+            self.marionette.execute_script("""
+              arguments[0].hidePopup();
+            """, script_args=[self.popup])
+        else:
+            (self.marionette.find_element(By.ID, 'identity-popup')
+                            .send_keys(Keys.ESCAPE))
```
Comment (whimboo) [inline firefox_puppeteer/ui/toolbars.py]: This is just `self.popup.send_keys()`.

File: firefox_puppeteer/ui/toolbars.py [hunk 3]
```diff
@@ -343,6 +343,25 @@ def selected_index(self):
...
+        Wait(self.marionette).until(
+            lambda _: not self.is_open)
```
Comment (whimboo) [inline firefox_puppeteer/ui/toolbars.py]: Please put this all in a single line. There is enough free space for it. Also please add a blank line above to make it more obvious that it doesn't belong to the else case.

**Comment by whimboo** [comment]
@Ethcelon please also file a bug on Bugzilla for this issue. So we can do the usual review process. Thanks.

**Comment by Ethcelon** [comment]
@whimboo could you review this?

File: firefox_puppeteer/ui/toolbars.py [hunk 4]
```diff
@@ -446,3 +446,21 @@ def verifier(self):
         :returns: Reference to the identity-popup content verifier.
         """
         return self.marionette.find_element(By.ID, 'identity-popup-content-verifier')
+
+    def close(self, force=False):
+        """Closes the identity popup.
+
+        :param force: If true, the popup is closed by its own hide function,
+                      otherwise a key event is sent to close the popup.
```
Comment (whimboo) [inline firefox_puppeteer/ui/toolbars.py]: I would shorten the description to "If True, the popup will be force closed". The second part you can add to the method summary like "... popup by hitting the escape key".

Comment (whimboo) [inline firefox_puppeteer/ui/toolbars.py]: Oh and please check e.g. the prefs library how the description for an optional parameter has to look like.

File: firefox_puppeteer/ui/toolbars.py [hunk 5]
```diff
@@ -446,3 +446,21 @@ def verifier(self):
...
+        """
+        if not self.is_open:
+            return
+
+        if force:
+            self.marionette.execute_script(
+                "arguments[0].hidePopup();",
+                script_args=[self.popup])
```
Comment (whimboo) [inline firefox_puppeteer/ui/toolbars.py]: Please also check the prefs library for the correct formatting of code for `execute_script`.

**Comment by whimboo** [comment]
@Ethcelon do you have an update for us?

**Comment by whimboo** [comment]
This PR will be finalized with PR #115.

**Pull request closed without merging**